File: app/src/main/java/org/worshipsongs/adapter/SongContentLandScapeViewerPageAdapter.java
package org.worshipsongs.adapter;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import org.worshipsongs.CommonConstants;
import org.worshipsongs.WorshipSongApplication;
import org.worshipsongs.service.SongService;
import org.worshipsongs.domain.Song;
import org.worshipsongs.fragment.SongContentLandscapeViewFragment;
import org.worshipsongs.service.AuthorService;

import java.util.List;

/**
 * author:madasamy
 * version:2.1.0
 */
public class SongContentLandScapeViewerPageAdapter extends FragmentStatePagerAdapter
{
    private String title;

    private SongService songService = new SongService(WorshipSongApplication.getContext());
    private AuthorService authorService = new AuthorService(WorshipSongApplication.getContext());
    private List<String> contents;
    private String authorName;
    private Song song;


    public SongContentLandScapeViewerPageAdapter(FragmentManager fragmentManager, String title)
    {
        super(fragmentManager);
        this.title = title;
        initSetUp();
    }

    public void initSetUp()
    {
        song = songService.findContentsByTitle(title);
        contents = song.getContents();
        authorName = authorService.findAuthorNameByTitle(title);
    }

    @Override
    public Fragment getItem(int position)
    {
        SongContentLandscapeViewFragment songContentLandscapeViewFragment = new SongContentLandscapeViewFragment();
        Bundle bundle = new Bundle();
        String content = contents.get(position);
        bundle.putString("content", content);
        bundle.putString(CommonConstants.TITLE_KEY, title);
        bundle.putString("authorName", authorName);
        bundle.putString("position", String.valueOf(position));
        bundle.putString("size", String.valueOf(contents.size()));
        bundle.putString("chord", song.getChord());
        songContentLandscapeViewFragment.setArguments(bundle);
        return songContentLandscapeViewFragment;
    }

    @Override
    public CharSequence getPageTitle(int position)
    {
        return "";
    }

    @Override
    public int getCount()
    {
        return contents.size();
    }
}

File: app/src/main/java/org/worshipsongs/adapter/SongContentLandScapeViewerPageAdapter.kt
package org.worshipsongs.adapter

import android.os.Bundle

import android.util.Log
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

import org.worshipsongs.CommonConstants
import org.worshipsongs.WorshipSongApplication
import org.worshipsongs.activity.SongContentViewActivity
import org.worshipsongs.service.SongService
import org.worshipsongs.domain.Song
import org.worshipsongs.fragment.SongContentLandscapeViewFragment
import org.worshipsongs.service.AuthorService

/**
 * @author: Madasamy
 * @since: 2.1.0
 */
class SongContentLandScapeViewerPageAdapter(fragmentManager: FragmentManager, private val title: String) : FragmentStatePagerAdapter(fragmentManager)
{

    private val songService = SongService(WorshipSongApplication.context!!)
    private val authorService = AuthorService(WorshipSongApplication.context!!)
    private var contents: List<String>? = null
    private var authorName: String? = null
    private var song: Song? = null


    init
    {
        Log.i(SongContentLandScapeViewerPageAdapter::class.java.simpleName, "Song content land scape view")
        initSetUp()
    }

    fun initSetUp()
    {
        song = songService.findContentsByTitle(title)
        contents = song!!.contents
        Log.i(SongContentLandScapeViewerPageAdapter::class.java.simpleName, "Song content size" + contents!!.size)
        authorName = authorService.findAuthorNameByTitle(title)
    }

    override fun getItem(position: Int): Fragment
    {

        val songContentLandscapeViewFragment = SongContentLandscapeViewFragment()
        val bundle = Bundle()
        val content = contents!![position]
        bundle.putString("content", content)
        bundle.putString(CommonConstants.TITLE_KEY, title)
        bundle.putString("authorName", authorName)
        bundle.putString("position", position.toString())
        bundle.putString("size", contents!!.size.toString())
        bundle.putString("chord", song!!.chord)
        songContentLandscapeViewFragment.arguments = bundle
        return songContentLandscapeViewFragment
    }

    override fun getPageTitle(position: Int): CharSequence?
    {
        return ""
    }

    override fun getCount(): Int
    {
        return contents!!.size
    }
}
